### PR TITLE
CV-228 Features

### PR DESCRIPTION
**Pull request by elliotritchie**

### CODE REVIEW

File: src/SFA.DAS.Authorization/AuthorizationContext.cs [hunk 1]
```diff
@@ -30,5 +31,10 @@ public bool TryGet<T>(string key, out T value)
 
             return exists;
         }
+
+        public string GetDescription()
```
Comment (iangledhill) [inline src/SFA.DAS.Authorization/AuthorizationContext.cs]: Would there be any benefit in overriding ToString() for this?

File: src/SFA.DAS.Authorization.Features/AuthorizationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            _featureTogglesService = featureTogglesService;
+            _logger = logger;
+        }
+
+        public Task<AuthorizationResult> GetAuthorizationResult(IReadOnlyCollection<string> options, IAuthorizationContext authorizationContext)
+        {
+            var authorizationResult = new AuthorizationResult();
+
+            if (options.Any())
```
Comment (iangledhill) [inline src/SFA.DAS.Authorization.Features/AuthorizationHandler.cs]: General obvs - .Any() and .Count() are used in places where .Length or .Count properties could be used instead. The previous is more consistent but the latter is more performant. Personal preference - just an obvs.

File: src/SFA.DAS.Authorization/AuthorizationService.cs [hunk 3]
```diff
@@ -48,7 +48,7 @@ public async Task<AuthorizationResult> GetAuthorizationResultAsync(params string
             
             var authorizationResults = await Task.WhenAll(
                 from h in _handlers
-                let o = options.Where(o => o.Contains(h.Prefix)).Select(o => o.Replace(h.Prefix, "")).ToList()
```
[the remark this answers is not in the document]
Comment (elliotritchie) [reply src/SFA.DAS.Authorization/AuthorizationService.cs]: Yeah possibly, I personally like the prefix needing to be an exact match though. I'll leave it as it is for now and see if there's any issues with developers getting them wrong.

File: src/SFA.DAS.Authorization/AuthorizationContext.cs (hunk 1 above)
Comment (elliotritchie) [reply src/SFA.DAS.Authorization/AuthorizationContext.cs]: Agreed, fixed.

File: src/SFA.DAS.Authorization.Features/AuthorizationHandler.cs (hunk 2 above)
Comment (elliotritchie) [reply src/SFA.DAS.Authorization.Features/AuthorizationHandler.cs]: Agreed, fixed.

**Pull request merged**